src/main/cpp/src/BloomFilterJni.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cudf_jni_apis.hpp"
+#include "dtype_utils.hpp"
+#include "jni_utils.hpp"
+
+#include "bloom_filter.hpp"
+#include "utilities.hpp"
+
+extern "C" {
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_BloomFilter_put(JNIEnv* env,
+                                                                         jclass,
+                                                                         jlong bloom_filter,
+                                                                         jlong bloom_filter_bytes,
+                                                                         long bloom_filter_bits,
```

Comment:
Curious why we're passing both the bit count and the byte count?  Seems like we could compute the byte count from the bit count trivially.

Reply:
Right. My thinking was to distinguish between the bit count as being an explicit property of the bloom filter itself and the byte count as just being a side-effect of how the underlying buffer ends up being represented.  Also, since we're passing the buffer pointer itself, it felt slightly more correct to include the byte count as being explicitly related.   

All that said, I'm not married to that at all.

Reply:
nit: technically bloom_filter_bits should be a `jlong` too.